.github/workflows/documentation.yml

```diff
@@ -13,13 +15,70 @@ jobs:
       - name: Test Build
         run: |
           if [ -e yarn.lock ]; then
-          yarn install
+            yarn install
+            yarn global add ts-node
           elif [ -e package-lock.json ]; then
-          npm ci
+            npm ci
+            npm install -g ts-node
           else
-          npm i
+            npm install
+            npm install -g ts-node
           fi
           npm run build
+
+      - name: Check if PR is from a fork
+        id: check_fork
+        run: |
+          if [ "${{ github.event.pull_request.head.repo.full_name }}" != "${{ github.repository }}" ]; then
+            echo "PR is from a fork"
+            echo "is_fork=true" >> $GITHUB_ENV
+          else
+            echo "PR is not from a fork"
+            echo "is_fork=false" >> $GITHUB_ENV
+          fi
+
+      - name: Run address checks
+        if: env.is_fork != 'true'
+        env:
+          ALCHEMY_API_KEY: ${{ secrets.ALCHEMY_API_KEY }}
+        run: |
+          ts-node scripts/runAddressChecks.ts
+
+      - name: confirm whether address checks passed
+        if: env.is_fork != 'true'
+        id: check_all_passed
+        run: |
+          V3_CHECKS_PASSED=$(jq -r '.addressChecks.allV3ChecksPassed' scripts/fetchedAddressData.json)
+          VEYFI_CHECKS_PASSED=$(jq -r '.addressChecks.allVeYfiChecksPassed' scripts/fetchedAddressData.json)
+          echo "V3 Checks Passed: $V3_CHECKS_PASSED"
+          echo "veYFI Checks Passed: $VEYFI_CHECKS_PASSED"
+          if [ "$V3_CHECKS_PASSED" = "true" ] && [ "$VEYFI_CHECKS_PASSED" = "true" ]; then
+            ALL_CHECKS_PASSED=true
+          else
+            ALL_CHECKS_PASSED=false
+          fi
+          echo "All checks passed: $ALL_CHECKS_PASSED"
+          echo "all_checks_passed=$ALL_CHECKS_PASSED" >> $GITHUB_ENV
+          if [ "$ALL_CHECKS_PASSED" != "true" ]; then
+            echo "Generating issue content..."
+            echo ":robot::warning: **Automatic Address Checks have failed!** :warning::robot:" > issue_body.md 
+            echo "The following contracts have changed:" >> issue_body.md
+            jq -r '.addressChecks.failedChecks[]' scripts/fetchedAddressData.json | while read CHECK; do
+                echo "- $CHECK" >> issue_body.md
```

Comment:
Sorry for taking longer to do this, i was tackling some bugs in real life and tasks on kong T.T.

So I guess we can simplify this whole check to be baked into the code itself, so we leave ci to just do the "devops" things. WDYT?


LMK if you need some help!

Reply:
Ah, so I would move the logic to determine whether the checks passed, as well as the content generation for the issue into the `runAddressChecks.ts` script? And then just read that in the issue creation step?

Reply:
Yeah, you can just do the checks inside the code and perform the issue creation there, without the need to "fail" ci to do that. So all the logic we have is inside of it. 

If you really prefer to sign to whoever sees the action history, you can simply throw as well after you created the issues. Or even `process.exit(1)`

Reply:
That makes sense. I don't want the CI to fail on this problem. Just make an issue to be fixed, so that makes sense. Will make that change today. Thanks!

Reply:
ok, I made those changes. Should make everything cleaner. Thanks for the help!